### PR TITLE
Feat(eos_designs): Change bgp_peer_groups to lower-case

### PR DESCRIPTION
**Pull request by ClausHolbechArista**
** Waiting for mpls design PR to merge, so we can cover those bgp peer groups with the new logic as well. **

## Change Summary

Change bgp_peer_groups to lower-case

## Component(s) name

`arista.avd.eos_designs`

## Proposed changes

- [x] Change the datamodel for `bgp_peer_groups` to only use lower-case keys
- [x] Gather the logic and defaults in switch facts instead of distributed everywhere
- [x] Ensure changes are backward compatible
- [x] Update release notes with the data model change
- [x] Update readme
- [x] Update most molecule scenarios to the new standard, but keep some with the old style

## How to test

No changes to molecule

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_designs/templates/overlay/ibgp/peer-groups.j2 [hunk 1]
```diff
@@ -29,8 +28,8 @@
       update_source: Loopback0
       remote_as: "{{ bgp_as }}"
       bfd: true
-{%     if bgp_peer_groups.rr_overlay_peers.password is arista.avd.defined() %}
-      password: "{{ bgp_peer_groups.rr_overlay_peers.password }}"
+{%     if switch.bgp_peer_groups.rr_overlay_peers.password is arista.avd.defined() %}
+      password: "{{ switch.bgp_peer_groups.rr_overlay_peers.password }}"
       send_community: all
       maximum_routes: 0
```
Comment (tgodaA) [inline ansible_collections/arista/avd/roles/eos_designs/templates/overlay/ibgp/peer-groups.j2:34]: Not related to this PR. But lines 20,21 are producing the same output as lines 33,34.

File: ansible_collections/arista/avd/roles/eos_designs/doc/fabric-variables.md [hunk 2]
```diff
@@ -70,19 +70,20 @@ bgp_ecmp: < number_of_ecmp_paths | default -> 4 >
...
 bgp_peer_groups:
-  IPv4_UNDERLAY_PEERS:
+   # Old mixed case key "IPv4_UNDERLAY_PEERS" is supported for backward-compatibility
+  ipv4_underlay_peers:
     name: < name of peer group | default -> IPv4-UNDERLAY-PEERS >
     password: "< encrypted password >"
-  MLAG_IPv4_UNDERLAY_PEER:
+   # Old mixed case key "MLAG_IPv4_UNDERLAY_PEER" is supported for backward-compatibility
+  mlag_ipv4_underlay_peer:
       name: < name of peer group | default -> MLAG-IPv4-UNDERLAY-PEER >
```
Comment (tgodaA) [inline ansible_collections/arista/avd/roles/eos_designs/doc/fabric-variables.md]: Please set the indentation similar to above peer group in lines 83, 84, 87, 88

**Review by tgodaA — APPROVED** [review]
LGTM

**Review by xaviramon — APPROVED** [review]
LGTM

**Pull request merged**